internal/workflow.go

```diff
@@ -549,20 +554,47 @@ func Sleep(ctx Context, d time.Duration) (err error) {
 	return
 }
 
-// RequestCancelWorkflow can be used to request cancellation of an external workflow.
+// RequestCancelExternalWorkflow can be used to request cancellation of an external workflow.
 // Input workflowID is the workflow ID of target workflow.
 // Input runID indicates the instance of a workflow. Input runID is optional (default is ""). When runID is not specified,
 // then the currently running instance of that workflowID will be used.
 // By default, the current workflow's domain will be used as target domain. However, you can specify a different domain
 // of the target workflow using the context like:
 //	ctx := WithWorkflowDomain(ctx, "domain-name")
-func RequestCancelWorkflow(ctx Context, workflowID, runID string) error {
+// RequestCancelExternalWorkflow return Future with failure or empty success result.
+func RequestCancelExternalWorkflow(ctx Context, workflowID, runID string) Future {
+	childWorkflowOnly := false // this means we are not limited to child workflow
+	return requestCancelExternalWorkflow(ctx, workflowID, runID, childWorkflowOnly)
+}
+
+func requestCancelExternalWorkflow(ctx Context, workflowID, runID string, childWorkflowOnly bool) Future {
 	ctx1 := setWorkflowEnvOptionsIfNotExist(ctx)
 	options := getWorkflowEnvOptions(ctx1)
-	if options.domain == nil {
-		return errors.New("need a valid domain")
+	future, settable := NewFuture(ctx1)
+
+	if options.domain == nil || *options.domain == "" {
+		settable.Set(nil, errDomainNotSet)
+		return future
+	}
+
+	if workflowID == "" {
+		settable.Set(nil, errWorkflowIDNotSet)
+		return future
 	}
-	return getWorkflowEnvironment(ctx).RequestCancelWorkflow(*options.domain, workflowID, runID)
+
+	resultCallback := func(result []byte, err error) {
+		settable.Set(result, err)
+	}
+
+	getWorkflowEnvironment(ctx).RequestCancelExternalWorkflow(
+		*options.domain,
+		workflowID,
+		runID,
+		false, // this means we are not limited to child workflow
```

Comment:
hard code? should it be the childWorkflowOnly argument?

Reply:
there are 2 API, one is childWorkflowFuture.RequestCancelChildWorkflow
another is workflow.RequestCancelExternalWorkflow

this is RequestCancelExternalWorkflow, so hard coded to false

Reply:
i don't get it. then why do you need the childWorkflowOnly pass in as parameter?

Reply:
childWorkflowOnly is used on the server side for this case:
parent workflow want to cancel a workflow if and only if the target workflow is its child workflow.